adafruit_magtag/magtag.py

```diff
@@ -293,6 +300,54 @@ def set_text(self, val, index=0, auto_refresh=True):
         if auto_refresh:
             self.refresh()
 
+    def exit_and_enter_deep_sleep(self, sleep_time):
+        """
+        Stops the current program and enters deep sleep. The program is restarted from the beginning
+        after a certain period of time. If USB is plugged in, it won't enter a true deep sleep.
```

Comment:
It's probably worth being more precise about USB. If you power it via a USB power pack, then it will truly deep sleep (or light sleep), because it's not connected (not doing CIRCUITPY or a serial connection).

Reply:
Ah, ok. Now I understand better.

Reply:
I'm not sure we need to reveal the USB detail. For a high level program, it'll work the same.

Reply:
I was thinking it might be useful to know about the power draw when connected or not, but maybe that's not so important.

Reply:
if someone is trying to measure the power draw via a USB-based meter, they will be disappointed (or more appropriately, they will get unexpected results)

Reply:
Ok, what would you like me to change?

Reply:
@dhalbert My worry is that there is a lot of subtly to it and here is not the best place to document it.

@makermelissa My proposal would be to link to the readthedocs for the alarm module for specifics. https://circuitpython.readthedocs.io/en/latest/shared-bindings/alarm/index.html

I think you could rename this to `exit_and_deep_sleep` to match the lower level api too.

Reply:
@tannewt, I changed it to the link, which makes sense. I also changed to your requested function name.

Reply:
I like the idea of a link as well, instead of just summarizing.